### ISSUE TITLE
changelog updates

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#1360]
I'm updating the changelog and bumping the version to 0.3.5 today. No special reason, just needs to be done. The tenative log is below, let me know if anything needs to be added.

```
### 0.3.5 - 2015-06-11
 
This patch improves overall stability and performance
 
* added 'object patch' and 'object new' commands @whyrusleeping
* improved symmetric NAT avoidance @jbenet
* move util.Key to blocks.Key @whyrusleeping
* fix memory leak in provider store @whyrusleeping
* updated webui to 0.2.0 @krl
* improved bitswap performance @whyrusleeping
* update fuse lib @cryptix
* fix path resolution @wking
* implement test_seq() in sharness @chriscool
* improve parsing of stdin for commands @chriscool
* fix 'ipfs refs' failing silently @whyrusleeping
* fix serial dialing bug @jbenet
* improved testing @chriscool @rht @jbenet
* fixed domain resolving @luzifer
* fix parsing of unwanted stdin @lgierth
* added CORS handlers to gateway @NodeGuy
* added `ipfs daemon --unrestricted-api` option @krl
* general cleanup of dependencies
 
### 0.3.4 - 2015-05-10
 
* fix ipns append bug @whyrusleeping
* fix out of memory panic @whyrusleeping
* add in expvar metrics @tv42
* bitswap improvements @whyrusleeping
* fix write-cache in blockstore @tv42
* vendoring cleanup @cryptix
* added `launchctl` plist for OSX @grncdr
* improved Dockerfile, changed root and mount paths @ehd
* improved `pin ls` output to show types @vitorbaptista
```

**Comment by jbenet** [comment]
add: 

```
### 0.3.5

* improved testing @chriscool @rht @jbenet
* fixed domain resolving @luzifer
* fix parsing of unwanted stdin @lgierth
* added CORS handlers to gateway @NodeGuy
* added `ipfs daemon --unrestricted-api` option @krl

### 0.3.4

* added `launchctl` plist for OSX @grncdr
* improved Dockerfile, changed root and mount paths @ehd
* improved `pin ls` output to show types @vitorbaptista
```

**Comment by whyrusleeping** [comment]
@jbenet updated my comment

**Comment by jbenet** [comment]
@whyrusleeping LGTM. there's many other contributions (thanks!!) but they're more repo-oriented so users dont need to be aware of them

**Comment by timgws** [comment]
:+1: I feel like 0.3.5 is going to be the best release so far!

**Comment by wking** [comment]
Looks like I missed the cutoff for this, but for what it's worth
here's what I had for sorting and linking the v0.3.5 changes:

This patch improves overall stability and performance.

#### User interface

* fix path resolution @wking @luzifer @whyrusleeping, #1208, #1272,
  #1275, #1233, #1235.  Migration notes in #1307.
* fixed domain resolving @luzifer, #1272
* added 'object patch' and 'object new' commands @whyrusleeping, #1134
* improved symmetric NAT avoidance @jbenet, #1342
* updated webui to 0.2.0 @krl, #1314
* added CORS handlers to gateway @NodeGuy, #1215
* improve parsing of stdin for commands @lgierth @chriscool, #1238,
  #1250, #1255, #1263, #1285
* fix 'ipfs refs' failing silently @whyrusleeping, #1254
* fix serial dialing bug @jbenet ??
* added `ipfs daemon --unrestricted-api` option @krl, #1221

#### Stability and performance

* improved bitswap performance @whyrusleeping, #1218, #1278, #1281,
  #1290
* fix memory leak in provider store @whyrusleeping, #1315
* parallelize crypto handshake @jbenet, #1227

#### Internal changes

* move util.Key to blocks.Key @whyrusleeping, #1318
* update fuse lib @cryptix, #1301
* implement test_seq() in sharness @chriscool, #1329
* improved testing @chriscool @whyrusleeping @rht @jbenet, #1235,
  #1293, #1310, #1341
* general cleanup of dependencies @whyrusleeping @cryptix @chriscool,
  #1269, #1302, #1305, #1323, #1330, #1339
* 'go vet' and golint, @rht, #1327

**Comment by jbenet** [comment]
:+1: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap:
